colcon_bundle/verb/_overlay_utilities.py

```diff
@@ -86,31 +94,46 @@ def create_dependencies_overlay(staging_path, overlay_path):
         dependencies_tar_gz_path
     ))
 
-    _generate_template(
-        'v2_setup.jinja2.sh',
-        'v2_setup.bash',
-        _CONTEXT_VAR_BASH
+    assets_directory = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        'assets'
+    )
+
+    shellscript_path = os.path.join(
+        assets_directory,
+        'v2_setup.sh'
+    )
+    shellscript_dest = os.path.join(
+        dependencies_staging_path,
+        'setup.sh'
+    )
+    shellscript_path_bash = os.path.join(
+        assets_directory,
+        'v2_setup.bash'
+    )
+    shellscript_dest_bash = os.path.join(
+        dependencies_staging_path,
+        'setup.bash'
     )
 
+    shutil.copy2(shellscript_path, shellscript_dest)
+    os.chmod(shellscript_dest, 0o755)
+
     _generate_template(
         'v2_setup.jinja2.sh',
-        'v2_setup.sh',
+        shellscript_dest,
         _CONTEXT_VAR_SH
     )
 
-    assets_directory = os.path.join(
-        os.path.dirname(os.path.realpath(__file__)), 'assets')
-    shellscript_path = os.path.join(assets_directory, 'v2_setup.sh')
-    shutil.copy2(shellscript_path,
-                 os.path.join(dependencies_staging_path, 'setup.sh'))
-    os.chmod(os.path.join(dependencies_staging_path, 'setup.sh'), 0o755)
-    shellscript_path_bash = os.path.join(
-        assets_directory,
-        'v2_setup.bash'
+    shutil.copy2(shellscript_path_bash, shellscript_dest_bash)
```

Comment:
This copy operation and the permissions setting is erased by the following template rendering write - it's a no-op